app/src/main/java/com/example/mystore/ui/login/LoginScreen.kt

```diff
@@ -31,30 +33,36 @@ import androidx.compose.ui.text.input.PasswordVisualTransformation
 import androidx.compose.ui.text.input.VisualTransformation
 import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.dp
-import com.example.mystore.hasDigits
-import com.example.mystore.isMixedCase
-import com.example.mystore.isOverEightCharacters
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.example.mystore.util.hasDigits
+import com.example.mystore.util.isOverSevenCharacters
+import com.example.mystore.model.LoginModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.json.JSONObject
+
 
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun LoginScreen(navigateToProductDetailsScreen: () -> Unit = {}) {
     var emailField by remember { mutableStateOf("") }
     var isEmailValid by remember { mutableStateOf(false) }
     var passwordField by remember { mutableStateOf("") }
-    var passwordVisible = rememberSaveable { mutableStateOf(false) }
-    val rightPassword = "Password01"
-    val rightEmail = "test@abv.bg"
+    val passwordVisible = rememberSaveable { mutableStateOf(false) }
     var isValidInput by remember { mutableStateOf(false) }
+    var isButtonEnabled by remember { mutableStateOf(true) }
     val mContext = LocalContext.current
+    val coroutineScope = rememberCoroutineScope()
+    var loginViewModel: LoginViewModel = hiltViewModel()
+    val errorMessage = "Your email or password are incorrect"
     val isValidEmail: (email: String) -> Boolean = { email ->
         android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches()
```

Comment:
You can readjust your import up top in the file so that you refer to this simply as `Patterns.EMAIL_ADDRESS` or even shorter - just `EMAIL_ADDRESS`. It all depends on how deep the import goes.